discord_bot.py

```diff
@@ -102,6 +86,22 @@ def __init__(self, root_save_folder: Path, config: RubberDuckConfig):
             (cc.get('name') or cc.get('id')): cc
             for cc in config['channels']
         }
+        self._retry_config = RetryConfig(max_retries=self._config["max_retries"], delay=self._config["delay"],backoff=self._config["backoff"])
```

Comment:
Once those settings are in a sub object, this can simply be `self._retry_config = self._config['retry_settings']`